assets/app/scripts/controllers/edit/buildConfig.js

```diff
@@ -0,0 +1,561 @@
+        "title": "--- None ---"
+      }
+    ];
+    $scope.breadcrumbs = [
+      {
+        title: "Builds",
+        link: "project/" + $routeParams.project + "/browse/builds"
+      },
+      {
+        title: $routeParams.buildconfig,
+        link: "project/" + $routeParams.project + "/browse/builds/" + $routeParams.buildconfig
+      },
+      {
+        title: "Edit"
+      }
+    ];
+    $scope.buildFrom = {
+      projects: [],
+      imageStreams: [],
+      tags: {},
+    };
+    $scope.pushTo = {
+      projects: [],
+      imageStreams: [],
+      tags: {},
+    };
+    $scope.sources = {
+      "binary": false,
+      "dockerfile": false,
+      "git": false,
+      "image": false,
+      "contextDir": false
+    };
+    $scope.triggers = {
+      "webhook": false,
+      "imageChange": false,
+      "configChange": false
+    };
+
+    AlertMessageService.getAlerts().forEach(function(alert) {
+      $scope.alerts[alert.name] = alert.data;
+    });
+    AlertMessageService.clearAlerts();
+    var watches = [];
+
+    ProjectsService
+      .get($routeParams.project)
+      .then(_.spread(function(project, context) {
+        $scope.project = project;
+        DataService.get("buildconfigs", $routeParams.buildconfig, context).then(
+          // success
+          function(buildConfig) {
+            $scope.buildConfig = buildConfig;
+            $scope.updatedBuildConfig = angular.copy($scope.buildConfig);
+            $scope.buildStrategy = $filter('buildStrategy')($scope.updatedBuildConfig);
+            $scope.strategyType = $scope.buildConfig.spec.strategy.type;
+            $scope.envVars = $filter('envVarsPair')($scope.buildStrategy.env);
+
+            angular.forEach($scope.buildConfig.spec.triggers, function(value, key) {
+              switch (value.type) {
+                case "Generic":
+                  break;
+                case "GitHub":
+                  $scope.triggers.webhook = true;
+                  break;
+                case "ImageChange":
+                  $scope.triggers.imageChange = true;
+                  break;
+                case "ConfigChange":
+                  $scope.triggers.configChange = true;
+                  break;
+              }
+            });
+            $scope.pickedTriggers = $scope.triggers;
+
+            angular.forEach($scope.buildConfig.spec.source, function(value, key) {
+              switch (key) {
+                case "binary":
+                  $scope.sources.binary = true;
+                  break
+                case "dockerfile":
+                  $scope.sources.dockerfile = true;
+                  break
+                case "git":
+                  $scope.sources.git = true;
+                  break
+                case "image":
+                  $scope.sources.image = true;
+                  break
+                case "contextDir":
+                  $scope.sources.contextDir = true;
+                  break
+              }
+            });
+
```

Comment:
Could some of this be pushed up into a service as well?
```javascript
$scope.options.pickedBuild = BuildConfigService.buildFromVars(buildConfig);  //  perhaps return an object like {from: {type:'',namespace:'',imageStream:'',dockerImage:""}}
$scope.options.pickedPush = BuildConfigService.pushToVars(buildConfig); 
```
I oversimplified in my sketch code, but it could be nice to reduce some logic in controller here.